src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -47,8 +47,41 @@ object Simplify {
     }
   }
 
+  private[this] def storeRepartitionability(
+    x: BaseIR,
+    repartitionability: Memo[(Boolean, Boolean)],
+    downstreamOK: Boolean
+  ): Unit = {
+    if (repartitionability.get(x).forall(_._1 != downstreamOK)) {
+      val selfOK = x.children.forall {
```

Comment:
`selfOK` can never change once stored, can it?

Reply:
true. I can do a getOrElse thing on this to avoid recalculating.